### PR TITLE
fix: parallel bitext tasks scores

### LINKED ISSUE #1018: NTREXBitextMining KeyError
```bash
ERROR:mteb.evaluation.MTEB:Error while evaluating NTREXBitextMining: 'sna_Latn-est_Latn'
Traceback (most recent call last):
  File "/env/lib/conda/gritkto/bin/mteb", line 8, in <module>
    sys.exit(main())
  File "/data/niklas/mteb/mteb/cli.py", line 370, in main
    args.func(args)
  File "/data/niklas/mteb/mteb/cli.py", line 118, in run
    eval.run(
  File "/data/niklas/mteb/mteb/evaluation/MTEB.py", line 388, in run
    raise e
  File "/data/niklas/mteb/mteb/evaluation/MTEB.py", line 367, in run
    mteb_task_result = MTEBResults.from_task_results(
  File "/data/niklas/mteb/mteb/load_results/mteb_results.py", line 89, in from_task_results
    eval_langs = subset2langscripts[hf_subset]
KeyError: 'sna_Latn-est_Latn'
```

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
Due to the implementation of parallel bitext tasks it computes all combinations even though only some are specified by eval_langs. This leads to a minor computational overhead when evaluating and most notably an error when constructing MTEBResults.

This fix does the following:
- it ensures that only desired language pairs are being evaluated
- it simplified bitext evaluation (there is no longer two cases)
- it additionally only encodes the sentence pairs required for the specific selected pairs (make it faster when we only want to consider a subset of languages)

fixes #1018 

I tested it using:
- A monolingual bitext task
- A multilingual parallel bitext task
- A multilingal non-parallel bitext task

All pass and reproduce the same results (cheers!)

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

File: mteb/abstasks/AbsTaskBitextMining.py [hunk 1]
```diff
@@ -62,8 +62,16 @@ def evaluate(self, model, split, **kwargs) -> dict[HFSubset, ScoresDict]:
     def _evaluate_subset(
         self, model, data_split: Dataset, parallel=False, **kwargs
     ) -> ScoresDict:
+        if parallel:
+            pairs = [langpair.split("-") for langpair in self.hf_subsets]
+        else:
+            pairs = [("sentence1", "sentence2")]
```
Comment (isaac-chung) [inline mteb/abstasks/AbsTaskBitextMining.py:68]: nit: simplify.
```suggestion
        pairs = [("sentence1", "sentence2")]
        if parallel:
            pairs = [langpair.split("-") for langpair in self.hf_subsets]
```

File: mteb/evaluation/evaluators/BitextMiningEvaluator.py [hunk 2]
```diff
@@ -38,32 +42,31 @@ def __call__(self, model: Encoder):
...
+        ]
+        n_subsets = len(subsets)
+
         embeddings = {}
-        for sub in tqdm.tqdm(
-            self.subsets, desc=f"Encoding {n_subsets}x{self.n} sentences"
-        ):
+        for sub in tqdm.tqdm(subsets, desc=f"Encoding {n_subsets}x{self.n} sentences"):
             embeddings[sub] = model_encode(
                 self.sentences[sub], model=model, prompt_name=self.task_name
```
Comment (isaac-chung) [inline mteb/evaluation/evaluators/BitextMiningEvaluator.py:54]: Do we need batch_size here as well?

File: mteb/evaluation/evaluators/BitextMiningEvaluator.py [hunk 3]
```diff
@@ -15,16 +15,20 @@
...
 
 
+DEFAULT_PAIR = [("sentence1", "sentence2")]
+
+
 class BitextMiningEvaluator(Evaluator):
-    def __init__(self, sentences, task_name: str | None = None, subsets=None, **kwargs):
+    def __init__(
+        self,
+        sentences,
```
Comment (isaac-chung) [inline mteb/evaluation/evaluators/BitextMiningEvaluator.py:24]:
```suggestion
        sentences: Dataset,
```

**Review by isaac-chung — APPROVED** [review]
Looks good! A few non-blocking suggestions.

File: mteb/evaluation/evaluators/BitextMiningEvaluator.py (hunk 3 above)
Comment (KennethEnevoldsen) [reply mteb/evaluation/evaluators/BitextMiningEvaluator.py:24]: Can't it also be a datasetDict? (generally only type hint when I am 100% sure)

**Comment by KennethEnevoldsen** [comment]
Since the issues are non-blocking I will just merge - I will add changes from discussion in a seperate PR

**Review by loicmagne — APPROVED** [review]
LGTM thanks @KennethEnevoldsen for fixing this, I forgot to handle this case in my initial implementation

**Pull request merged**